LLMonFHIR/FHIR Display/OpenAIChatView.swift

```diff
@@ -87,4 +77,112 @@ struct OpenAIChatView: View {
+            ChatFunctionDeclaration(
+                name: "get_resource_titles",
+                description: String(localized: "FUNCTION_DESCRIPTION"),
+                parameters: JSONSchema(
+                    type: .object,
+                    properties: [
+                        "resources": .init(type: .string, description: String(localized: "PARAMETER_DESCRIPTION"), enumValues: stringResourcesArray)
+                    ],
+                    required: ["resources"]
+                )
+            )
+        ]
+
+        let chat = [
+            Chat(role: .user, content: chat.last?.content ?? ""),
+            Chat(role: .system, content: String(localized: "FUNCTION_CONTEXT") + stringResourcesArray.rawValue)
+        ]
+        
+        let chatStreamResults = try await openAPIComponent.queryAPI(withChat: chat, withFunction: functions)
+        
+        class ChatFunctionCall {
+            var name: String = "", arguments: String = "", finishReason: String = ""
+        }
+        
+        var functionCall = ChatFunctionCall()
+        
+        for try await chatStreamResult in chatStreamResults {
+            for choice in chatStreamResult.choices {
+                if let deltaName = choice.delta.name {
+                    functionCall.name += deltaName
+                }
+                if let deltaArguments = choice.delta.functionCall?.arguments {
+                    functionCall.arguments += deltaArguments
+                }
+                if let finishReason = choice.finishReason {
+                    functionCall.finishReason += finishReason
+                    if finishReason == "get_resource_titles" { break }
+                }
+            }
+        }
+        
+        var functionCallOutputArray = [String]()
+        
+        if functionCall.finishReason == "function_call" {
+            let trimmedArguments = functionCall.arguments.trimmingCharacters(in: .whitespacesAndNewlines)
+            
+            if let resourcesRange = trimmedArguments.range(of: "\"resources\": \"([^\"]+)\"", options: .regularExpression) {
+                functionCallOutputArray = trimmedArguments[resourcesRange]
+                    .replacingOccurrences(of: "\"resources\": \"", with: "")
+                    .replacingOccurrences(of: "\"", with: "")
+                    .components(separatedBy: ",")
+            }
+        }
+        
+        return functionCallOutputArray
+    }
+
+    private func processFunctionCallOutputArray(functionCallOutputArray: [String], resourcesArray: [FHIRResource]) {
+        let stringResourcesArray = resourcesArray.map { "\($0.displayName) in \($0.resourceType)" }
+        for resource in functionCallOutputArray {
+            var stringResource = resource
+            stringResource = resource.trimmingCharacters(in: .whitespaces)
+            if let index = stringResourcesArray.firstIndex(of: stringResource) {
+                let resourceJSON = resourcesArray[index].jsonDescription
```

Comment:
I think it might not be wise to purely trust on the fact that the indexes are the same and they stay the same across function calls.
We should rather filter the `resourcesArray` and try to get the resource, e.g., using the extension that I recommended in the other comment.